src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -465,9 +465,26 @@ private static void BindInstance(
             }
             else
             {
-                if (isParentCollection && bindingPoint.Value is null && string.IsNullOrEmpty(configValue))
+                // Reaching this point indicates that the configuration section is a leaf node with a string value.
+                // Typically, configValue will be an empty string if the value in the configuration is empty or null.
+                // While configValue could be any other string, we already know it cannot be converted to the required type, as TryConvertValue has already failed.
+
+                if (!string.IsNullOrEmpty(configValue))
+                {
+                    // If we have a value, but no children, we can't bind it to anything
+                    // We already tried calling TryConvertValue and couldn't convert the configuration value to the required type.
+                    if (options.ErrorOnUnknownConfiguration)
+                    {
+                        Debug.Assert(section is not null);
+                        throw new InvalidOperationException(SR.Format(SR.Error_FailedBinding, section.Path, type));
+                    }
+
+                    return;
```

Comment:
nit: would this read better if we made this an `else if` clause to the parent statement above?  That would avoid extra return.